File: typescript/gigasecond/gigasecond.ts
class Gigasecond {
  public actual: Date;

  constructor(date: Date) {
    this.actual = date;
  }

  date(): Date {
    const gigDate = Math.pow(10, 12);
    return new Date(this.actual.getTime() + gigDate);
  }
}

export default Gigasecond;

File: go/gigasecond/gigasecond.go
// Package gigasecond implements the gigasecond algorithm
package gigasecond

import (
	"math"
	"time"
)

// AddGigasecond calculates the time after pass one gigasecond and returns this value
func AddGigasecond(t time.Time) time.Time {
	gigaSecond := math.Pow(10.0, 9)

	then := gigaSecond + float64(t.Unix())

	t = time.Unix(int64(then), 0)

	return t
}
